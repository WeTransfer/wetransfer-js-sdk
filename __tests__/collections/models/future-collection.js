const { futureCollection } = require('../../../src/collections/models');

describe('Future collection normalizer', () => {
  describe('normalizeCollection function', () => {
    let collection = {};
    beforeEach(() => {
      collection = {
        name: 'WeTransfer rocks',
        description: '',
      };
    });

    it('should return a normalized transfer', () => {
      const normalized = futureCollection(collection);
      expect(normalized).toMatchSnapshot();
    });

    it('should remove extra properties', () => {
      const extraProps = Object.assign({}, collection, { date: new Date() });
      const normalized = futureCollection(extraProps);
      expect(normalized).toMatchSnapshot();
    });

    it('add default values', () => {
      const normalized = futureCollection({});
      expect(normalized).toMatchSnapshot();
    });
  });
});
