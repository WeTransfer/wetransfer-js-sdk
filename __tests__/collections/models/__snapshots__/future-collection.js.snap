// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Future collection normalizer normalizeCollection function add default values 1`] = `
Object {
  "description": "",
  "name": "",
}
`;

exports[`Future collection normalizer normalizeCollection function should remove extra properties 1`] = `
Object {
  "description": "",
  "name": "WeTransfer rocks",
}
`;

exports[`Future collection normalizer normalizeCollection function should return a normalized transfer 1`] = `
Object {
  "description": "",
  "name": "WeTransfer rocks",
}
`;
